aiida/backends/tests/cmdline/commands/test_process.py

```diff
@@ -219,6 +220,51 @@ def test_list(self):
             self.assertEqual(len(get_result_lines(result)), 1)
             self.assertEqual(get_result_lines(result)[0], str(self.calcs[0].pk))
 
+    def test_list_worker_slot_warning(self):
+        """
+        Test that the if the number of used worker process slots exceeds a threshold,
+        that the warning message is displayed to the user when running `verdi process list`
+        """
+        from aiida.engine import ProcessState
+
+        # Get the number of allowed processes per worker:
+        from aiida.manage.external.rmq import _RMQ_TASK_PREFETCH_COUNT
+        limit = int(_RMQ_TASK_PREFETCH_COUNT * 0.9)
+
+        # Create additional active nodes such that we have 90% of the active slot limit
+        # (including the 6 active nodes created in the class setup fixture)
+        for _ in six.moves.range(limit - 6):
+            calc = WorkFunctionNode()
+            calc.set_process_state(ProcessState.RUNNING)
+            calc.store()
+
+        # Override the call to the circus client to retrieve the number of workers
+        # As we don't have a running circus client, this will normally fail, so here we simulate the
+        # response by redefining the function to get the final value we want.
+        import aiida.cmdline.utils.common
+        real_get_num_workers = aiida.cmdline.utils.common.get_num_workers
+        aiida.cmdline.utils.common.get_num_workers = lambda: 1
+
+        # Default cmd should not throw the warning as we are below the limit
+        result = self.cli_runner.invoke(cmd_process.process_list)
+        self.assertIsNone(result.exception, result.output)
+        warning_phrase = "of the available daemon worker slots have been used!"
+        self.assertTrue(all([warning_phrase not in line for line in get_result_lines(result)]))
+
+        # Add one more running node to put us over the limit
+        calc = WorkFunctionNode()
+        calc.set_process_state(ProcessState.RUNNING)
+        calc.store()
+
+        # Now the warning should fire
+        result = self.cli_runner.invoke(cmd_process.process_list)
+        self.assertIsNone(result.exception, result.output)
+        warning_phrase = "% of the available daemon worker slots have been used!"
+        self.assertTrue(any([warning_phrase in line for line in get_result_lines(result)]))
+
+        # Reset the redifined function
+        aiida.cmdline.utils.common.get_num_workers = real_get_num_workers
```

Comment:
I am afraid that this might still be problematic, if the tests excepts before it gets here. I think the safest think would be to separate this test in its own class, so you can properly change the `get_num_workers` function in the `setUp` and `tearDown`. Sorry to be asking yet another change, but the branch anyway had to be brought up to date, with develop. I promise this is the last change I ask for :)

Reply:
Ha, no problem. 
I've not refactored this test in to it's own class, and we do the method overloading in the setup and teardown.

Reply:
*not* should be *now* ? :wink:

Reply:
Brutal